### PR TITLE
Improve usability of IHtmlLocalizer & associated API

### PR DESCRIPTION
**Pull request by DamianEdwards**
- IHtmlLocalizer no longer derives from IStringLocalizer
- IHtmlLocalizer indexer now returns LocalizedHtmlString
- IHtmlLocalizer has GetString methods now that act the same as IStringLocalizer.GetString
- Made LocalizedHtmlString a struct to match LocalizedString
- Updated samples in response to aspnet/Localization#167
- Fixes some doc comments
- Fixed tests
- #3716

@ryanbrandenburg @rynowak @lodejard

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Localization/IHtmlLocalizer.cs [hunk 1]
```diff
@@ -1,37 +1,66 @@
...
+
+        /// <summary>
+        /// Gets all string resources.
+        /// </summary>
+        /// <param name="includeAncestorCultures">
+        /// A <see cref="System.Boolean"/> indicating whether to include
+        /// strings from ancestor cultures.
+        /// </param>
+        /// <returns>The strings.</returns>
+        IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures);
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNet.Mvc.Localization/IHtmlLocalizer.cs]: includeParentCultures?

**Comment by ryanbrandenburg** [comment]
What are the implications of making LocalizedHtmlString a struct? I know we're trying to follow the pattern laid out, but I'm curious for my own knowledge

**Comment by ryanbrandenburg** [comment]
ctrl+f Ancestor->Parent as per the discussion on the other PR.

File: src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    /// Extension methods for <see cref="IHtmlLocalizer"/>.
+    /// </summary>
+    public static class HtmlLocalizerExtensions
+    {
+        /// <summary>
+        /// Gets the <see cref="LocalizedHtmlString"/> resource for a specific name.
+        /// </summary>
+        /// <param name="key">The key to use.</param>
+        /// <returns>The <see cref="LocalizedHtmlString"/> resource.</returns>
+        public static LocalizedHtmlString GetHtml(this IHtmlLocalizer htmlLocalizer, string name)
```
Comment (ryanbrandenburg) [inline src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizerExtensions.cs]: Much clearer than Html().

File: src/Microsoft.AspNet.Mvc.Localization/IHtmlLocalizer.cs (hunk 1 above)
Comment (DamianEdwards) [inline src/Microsoft.AspNet.Mvc.Localization/IHtmlLocalizer.cs]: :+1:

**Comment by DamianEdwards** [comment]
@ryanbrandenburg the implications are that you can no longer derive from it and it will be copied on passing.

:up: :date:

**Comment by ryanbrandenburg** [comment]
:shipit:

**Pull request merged**